source/dbo/Stored Procedures/Unmatched Specimens/uspGenerateUnmatchedSpecimensLinkedNotifications.sql

```diff
@@ -0,0 +1,66 @@
+﻿CREATE PROCEDURE [dbo].[uspGenerateUnmatchedSpecimensLinkedNotifications]
+AS
+BEGIN TRY
+
+	-- First populate RejectSpecimenUsers table
+	EXEC dbo.uspGenerateRejectSpecimenUsers
+
+	TRUNCATE TABLE [dbo].[UnmatchedSpecimensLinkedNotifications]
+	
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'Unmatch' AS EventType INTO #PreviousMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected'
+
+	SELECT us.ReferenceLaboratoryNumber, nsm.NotificationID, 'RejectPotentialMatch' AS EventType INTO #PreviousPossibleMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.NotificationSpecimenMatch nsm ON nsm.ReferenceLaboratoryNumber = us.ReferenceLaboratoryNumber
+	WHERE nsm.MatchType = 'Rejected-Possible'
+
+	SELECT mr.clientsourceID AS ReferenceLaboratoryNumber, mr.rclientsourceID AS NotificationID, 'PoorQualityMatch' AS EventType INTO #PoorQualityMatch
+	FROM UnmatchedSpecimens us
+	JOIN [$(NTBS_Specimen_Matching)].dbo.MatchResults mr ON mr.clientsourceID = us.ReferenceLaboratoryNumber
+	--Exclude any matches that have previously been rejected (otherwise every rejection has an accompanying match)
+	LEFT JOIN #PreviousMatch pm on pm.NotificationID = mr.rclientsourceID AND pm.ReferenceLaboratoryNumber = mr.clientsourceID
+	LEFT JOIN #PreviousPossibleMatch ppm on ppm.NotificationID = mr.rclientsourceID AND ppm.ReferenceLaboratoryNumber = mr.clientsourceID
+	WHERE pm.EventType IS NULL AND ppm.EventType IS NULL
+	AND mr.[weight] < 34.50
+
+	CREATE TABLE #AllMatches (ReferenceLaboratoryNumber NVARCHAR(50), NotificationID int, EventType NVARCHAR(50))
+	INSERT INTO #AllMatches SELECT * FROM #PreviousMatch
+	INSERT INTO #AllMatches SELECT * FROM #PreviousPossibleMatch
+	INSERT INTO #AllMatches SELECT * FROM #PoorQualityMatch
+
+	INSERT INTO [dbo].[UnmatchedSpecimensLinkedNotifications]
+	SELECT am.ReferenceLaboratoryNumber
+		,CASE
+			WHEN am.EventType = 'RejectPotentialMatch' THEN 'Possible match rejected by ' + u.UserDisplayName + ' on ' + FORMAT(u.RejectionTime, 'dd MMM yyyy')
+			WHEN am.EventType = 'Unmatch' THEN 'Confirmed match rejected by ' + u.UserDisplayName + ' on ' + FORMAT(u.RejectionTime, 'dd MMM yyyy')
+			WHEN am.EventType = 'PoorQualityMatch' THEN 'Poor quality potential match'
+		END AS NotificationLinkReason
+		,am.NotificationID
+		,n.NotificationStatus
+		,p.NhsNumber
+		,p.Dob AS BirthDate
+		,UPPER(p.FamilyName) + ', ' + p.GivenName AS [Name]
```

Comment:
Is this logic that is usually done at the database level?

Reply:
I'll move this into powerBI